compiler/src/dotty/tools/dotc/typer/TypeAssigner.scala

```diff
@@ -291,26 +291,28 @@ trait TypeAssigner {
 
   def assignType(tree: untpd.Super, qual: Tree, inConstrCall: Boolean, mixinClass: Symbol = NoSymbol)(implicit ctx: Context) = {
     val mix = tree.mix
-    val qtype @ ThisType(_) = qual.tpe
-    val cls = qtype.cls
-
-    def findMixinSuper(site: Type): Type = site.parents filter (_.name == mix.name) match {
-      case p :: Nil =>
-        p
-      case Nil =>
-        errorType(SuperQualMustBeParent(mix, cls), tree.pos)
-      case p :: q :: _ =>
-        errorType("ambiguous parent class qualifier", tree.pos)
+    qual.tpe match {
+      case err: ErrorType => untpd.cpy.Super(tree)(qual, mix).withType(err)
```

Comment:
Tree copier provides structural sharing and is used when you think that the tree might stay the same after your changes.